lua/cmp/utils/spec.lua

```diff
@@ -5,16 +5,14 @@ local types = require('cmp.types')
 local spec = {}
 
 spec.before = function()
-  vim.cmd [[
-    bdelete!
-    enew!
-    setlocal virtualedit=all
-  ]]
+	vim.api.nvim_buf_delete(0, {force = true})
+	vim.api.nvim_command [[enew!]]
+	vim.opt.virtualedit = 'all'
```

Comment:
Just to explain this… `virtualedit` has no local option. From `:setlocal`:

> Not all options have a local value.  If the option does not have a local	value the global value is set.

Therefore `:setlocal` was equivalent to `:set` here. Trying to `:setlocal` with Lua resulted in an error, which confirms it. So `vim.opt` will have to do.

Reply:
what about `vim.opt_local.virtualedit`?

Reply:
I tried that along with `bo` and `wo`. The problem is that `virtualedit` is a global option (see `:h 'virtualedit'`), so even doing `:setlocal` has no specific effect and just resolves to a `:set`.